twenty-first/src/storage/storage_schema/simple_rusty_storage.rs

```diff
@@ -0,0 +1,63 @@
+use super::super::level_db::DB;
+use super::enums::WriteOperation;
+use super::{traits::StorageWriter, DbtSchema, SimpleRustyReader};
+use crate::sync::AtomicRw;
+use leveldb::batch::WriteBatch;
+use std::sync::Arc;
+
+/// Database schema and tables logic for RustyLevelDB. You probably
+/// want to implement your own storage class after this example so
+/// that you can hardcode the schema in new(). But it is nevertheless
+/// possible to use this struct and add to the schema.
+pub struct SimpleRustyStorage {
+    /// dynamic DB Schema.  (new tables may be added)
+    pub schema: DbtSchema<SimpleRustyReader>,
+}
+
+impl StorageWriter for SimpleRustyStorage {
+    #[inline]
+    fn persist(&mut self) {
+        let write_batch = WriteBatch::new();
```

Comment:
It's very mysterious to me that `write_batch` does not have to be mutable. I looked through the call graph and saw that it's because `unsafe` is used internally, to call the c++ function, I think. Should we be worried about that use of `unsafe`?

Reply:
unsafe is used internally in both `leveldb` and `rs-leveldb`.   It's not unusual when dealing with C pointers and such in rust.

However, it is interesting to note that in `leveldb` WriteBatch::put() takes `&mut self`, but in `rs-leveldb` it takes `&self`.    I would need to dig into the projects commit history to (maybe) understand why.

The biggest thing that concerns me about the `rs-leveldb` crate so far is that when I was making the bench tests, I was getting some segfaults, I believe when writing.    I figured out this only occurred when passing a non default Cache object in the config `Options`, so I stopped doing that, and made a mental note to look into it later.  Haven't seen a segfault or other problem since.  Now that you point out this discrepancy, well it could possibly be related somehow.

My feeling is that we could merge this PR with `rs-leveldb`, and change it a bit later if decide we need to.  We aren't in production yet.   And rs-leveldb has otherwise been solid in my test usage.   I'm not sure how long it would take to switch to `leveldb`, but perhaps a couple days.

Reply:
Alright. Let's stickwith the default cache in options then